plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1290,17 +1421,28 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
             // As a precaution, clear the caches
             statisticsAccess.invalidateCache(schemaTableName, Optional.of(location));
             transactionLogAccess.invalidateCache(schemaTableName, Optional.of(location));
-            metastore.createTable(session, table, principalPrivileges);
+            if (handle.getReadVersion().isPresent()) {
+                metastore.replaceTable(session, table, principalPrivileges);
+            }
+            else {
+                metastore.createTable(session, table, principalPrivileges);
+            }
         }
         catch (Exception e) {
             // Remove the transaction log entry if the table creation fails
-            try {
-                Location transactionLogDir = Location.of(getTransactionLogDir(location));
-                fileSystemFactory.create(session).deleteDirectory(transactionLogDir);
+            if (!writeCommitted) {
```

Comment:
`writeCommited` logic could be added in  a _preparatory_ commit.

Reply:
But the writeCommited logic is very specific to CORTAS operation so I'm not sure if adding them in a preparatory commit would help us here. WDYT ?